src/gemm_v2.cpp

```diff
@@ -324,41 +325,216 @@ inline void ProfilingRecordStop(const Handle& handle, HipEventPtr& start, HipEve
     handle.ResetKernelTime();
     handle.AccumKernelTime(mS);
 }
+
+#if MIOPEN_USE_HIPBLASLT
+static inline void check_hipblas_status(hipblasStatus_t status)
+{
+    if(status != hipblasStatus_t::HIPBLAS_STATUS_SUCCESS)
+    {
+        MIOPEN_THROW(miopenStatusInternalError, "hipBlasLt error encountered");
+    }
+}
+
+template <typename DataTypeAB, typename DataTypeC>
+static void miopen_hipblasLt_gemm(const miopen::Handle& handle,
+                                  const miopen::GemmDescriptor& gemm_desc,
+                                  ConstData_t A,
+                                  std::size_t a_offset,
+                                  ConstData_t B,
+                                  std::size_t b_offset,
+                                  hipDataType hip_type_AB,
+                                  Data_t C,
+                                  std::size_t c_offset,
+                                  hipDataType hip_type_C)
+{
+    float alpha = gemm_desc.alpha;
+    float beta  = gemm_desc.beta;
+    hipblaslt_ext::GemmInputs inputs;
+    // Note: Need to const cast here due to hipblaslt_ext::GemmInputs API requirements.
+    // NOLINTBEGIN(cppcoreguidelines-pro-type-const-cast)
+    inputs.a = const_cast<DataTypeAB*>(static_cast<const DataTypeAB*>(A) + a_offset);
+    inputs.b = const_cast<DataTypeAB*>(static_cast<const DataTypeAB*>(B) + b_offset);
+    // NOLINTEND(cppcoreguidelines-pro-type-const-cast)
+    inputs.c     = static_cast<DataTypeC*>(C) + c_offset;
+    inputs.d     = static_cast<DataTypeC*>(C) + c_offset;
+    inputs.alpha = &alpha;
+    inputs.beta  = &beta;
+
+    // TODO - bharriso - Need to pass down workspace size & pointer.
```

Comment:
[R] I would use the following format:
```C++
/// \todo <...stuff...> --<githubUserId> <date>
```
If a user does not have a githubUserId, then provide his/her real name or email.

```suggestion
    /// \todo Need to pass down workspace size & pointer. --bharriso June 2024
```